src/PreserveModuleNamePlugin.ts

```diff
@@ -46,37 +59,43 @@ export class PreserveModuleNamePlugin {
             id = fixNodeModule(realModule, modulesBeforeConcat) || 
                  makeModuleRelative(roots, realModule.resource) ||
                  aliasRelative(alias, realModule.resource);
-                    
+
           if (!id)
             throw new Error(`Can't figure out a normalized module name for ${realModule.rawRequest}, please call PLATFORM.moduleName() somewhere to help.`);
-          
+
           // Remove default extensions 
           normalizers.forEach(n => id = id!.replace(n, ""));
-          
+
           // Keep "async!" in front of code splits proxies, they are used by aurelia-loader
           if (/^async[?!]/.test(realModule.rawRequest)) 
             id = "async!" + id;
-          
+
           id = id.replace(/\\/g, "/");
           if (module.buildMeta)  // meta can be null if the module contains errors
             module.buildMeta["aurelia-id"] = id;
-          if (!this.isDll)
-            module.id = id;
+
+          if (!this.isDll) {
+            compilation.chunkGraph.setModuleId(module, id);
```

Comment:
using this should help us being forward compatible with webpack v6, and `module.id` is deprecated